QualityControl/lib/services/QcdbDownload.service.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { LogManager } from '@aliceo2/web-ui';
+import { pipeline } from 'node:stream';
+import { promisify } from 'node:util';
+
+/**
+ * @class
+ * Class which sets up the QCDB download service.
+ */
+export class QcdbDownloadService {
+  /**
+   * Constructor
+   * @param {object} config - application's {config.js}.ccdb, needed for Qcdb hostname+port
+   */
+  constructor(config = {}) {
+    this._hostname = config.hostname ?? 'localhost';
+    this._port = config.port ?? 8080;
+    this._protocol = config.protocol ?? 'http';
+    this._target = `${this._protocol}://${this._hostname}:${this._port}`;
+
+    this._logger = LogManager.getLogger(`${process.env.npm_config_log_label ?? 'qcg'}/proxy`);
+    this._pipelineAsync = promisify(pipeline);
+  }
+
+  /**
+   * Get ROOT object from qcdb.
+   * If a response object is given it is assumed that this is the only request
+   * the body of the answer from qcdb will then be streamed into our response.
+   * @param {string} objectId - id of ROOT object to retrieve from qcdb.
+   * @param {Express.Response} res - Optional Express response object if we want to stream qcdb's answer as our own.
+   * @returns {File|boolean} - ROOT file from qcdb, false if error and true if it responded itself.
+   */
+  async requestObject(objectId, res = undefined) {
+    this._logger.infoMessage(`Object ID Request: ${objectId}`);
+    const qcdbRequest = new Request(`${this._target}/download/${objectId}`, { method: 'GET' });
+    try {
+      const response = await fetch(qcdbRequest);
```

Comment:
I would do:
```suggestion
      const response = await fetch(`${this._target}/download/${objectId}`);
```
for readability